linera-execution/src/lib.rs

```diff
@@ -783,9 +786,44 @@ pub enum Query {
     },
 }
 
+/// The outcome of the execution of a query.
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+pub struct QueryOutcome<Response = QueryResponse> {
+    pub response: Response,
+    pub operations: Vec<Operation>,
+}
+
+impl From<QueryOutcome<SystemResponse>> for QueryOutcome {
```

Comment:
Are we actually using this?

Reply:
Yes, [here](https://github.com/linera-io/linera-protocol/pull/3217/files/f4e9b4a21fa7d977f4b5774b7f48de40f766151b#diff-c8417387ee85ef169137900d33ff4ac909c49dce26129c49a994d748fc9c9c53R493).

Reply:
Ah, the force push removed it :/ Here's a more permanent link:

https://github.com/jvff/linera-protocol-archive/blob/13dbdf447c0f505f6d0644fb1cc47e0d29a4d659/linera-execution/src/execution.rs#L493

Reply:
So there's exactly one system query, and it returns the chain ID and balance? :thinking: